### DISCUSSION TITLE
XOPBOT

### DISCUSSION THREAD

**Discussion by brandontorreglosa** [brandontorreglosa/xopbot#1] [General]
<h1>Questions / Bugs Discussion #1</h1>
Have any questions about the bot's new version? Answer them here and our contributors will reply as soon as possible.
<br/>
If you have found any errors or bugs on / from the bot please comment down here <b><code>how and when</code></b> did it happen and we will reply as soon as possible.
<br/>
You can also leave some suggestions if wanted down here aswell but with a detail that it is supported in <b>v12</b> and not <b>v13</b>
<br/>
Any questions / bugs pointing to <b>beta commands</b> will not be answered, as beta commands are still through testing. They do work but are not added everywhere on each command, such as <b>command logging</b> that will be finished in the latest version.
<br/>
We will constantly be updating the bot so it is rather wise to <b>fork</b> the repository or either clone it. You can fork it by clicking <a href="https://github.com/HACKERPROTM/XOPBOT-24-7">here</a>
<br/>
Contributors: @HACKERPROTM @NIKOLIS3 @AryanDaga1 @Simpleboy353
<br/>
<a href="https://xopbot.glitch.me">Website</a>&nbsp;|&nbsp;<a href="https://xopbot.glitch.me/#donations">Donate</a>&nbsp;|&nbsp;<a  href="https://xopbot.glitch.me/services/api/home">API</a>
<hr /><em>This discussion was created from the release <a href='https://github.com/HACKERPROTM/XOPBOT-24-7/releases/tag/v60.0.03'>XOPBOT</a>.</em>
<em>The discussion only supports version <b><code>60.0.03</code></b> and higher versions of it. Lower versions are not supported since they are <b>deprecated</b>.</em>

**Comment by brandontorreglosa** [comment]
Comment by clicking the <b><code>💬 write comment</code></b> button. Define your problem or suggestion through the chatbox and we will mark it as a mark that is required from the comment. Please make sure to read the introduction of the discussion before commiting anything on the comment section as false commenting will result in a ban!

**Comment by dixlyn** [comment]
bro their are so many errors the bot wont even run for me [Cannot read properties of undefined (reading 'role.id')]

**Reply by brandontorreglosa** [reply]
Have you deployed to heroku? You can't locally deploy because, the version 12.5.3 of discord.js is deprecated and no longer is supported so moderation commands most likely fail, but if you deploy it with heroku it would work. Because heroku fixes deployment errors. If you need help in deploying it to heroku tell me. For now you could invite the bot if you want to check it out.

**Reply by brandontorreglosa** [reply]
Hey, I double-checked my code and i have encountered an error with the commands `addrole` and `removerole` where these errors are actually not associated with heroku. I Will find a fix as soon as possible and gonna run it through test's as for the other errors they are associated with npm and won't be shown if you deploy it to heroku. If you encounter another command that throws errors please contact me here.

**Reply by brandontorreglosa** [reply]
<p align="left" width="20%">@dixlyn Yeah, i tried different ways to fix it but I couldn't because as I explained in my first reply the package is deprecated although the other commands work that seem to don't. I will be updating the bot so I recommend you to just remove those two commands and host the bot to heroku until I update it to v13. Thank you 👋</p>
<img src="https://user-images.githubusercontent.com/84269666/171619767-c4db88b4-dcba-40ac-a355-c0249ed58d82.gif">

**Reply by brandontorreglosa** [reply]
@dixlyn, i fixed all possible errors now please try to deploy it. it took some time and got some help from StackOverflow but managed to fix it. as for the role commands there was a typo but I fixed it with some help

**Comment by brandontorreglosa** [comment]
XOPBOT will no longer be accessible for the public. So we are sadly closing any new updates from happening except new models from a side client. Thank you all for the support that you have given.

**Comment by brandontorreglosa** [comment]
CLOSED!!!
